gpu_direct/gpu.c

```diff
@@ -0,0 +1,326 @@
+  ZE_CHECK_ERROR(zeContextCreate(ctx->driverHandle, &ctxDesc, &ctx->deviceContext));
+
+  // Get decive count
+  ZE_CHECK_ERROR(zeDeviceGet(ctx->driverHandle, &ctx->deviceCount, NULL));
+  printf("Number of devices: %d\n", ctx->deviceCount);
+  if (deviceIndex >= ctx->deviceCount) {
+    fprintf(stderr, "Init error: provided device index is out of range\n");
+    return -1;
+  }
+
+  // Allocate and retrieve device handlers
+  ctx->devices = calloc(ctx->deviceCount, sizeof(ze_device_handle_t));
+  if (ctx->devices == NULL) {
+    fprintf(stderr, "Can't allocate memory for devices handlers\n");
+    free(ctx->drivers);
+    return -1;
+  }
+  ZE_CHECK_ERROR(zeDeviceGet(ctx->driverHandle, &ctx->deviceCount, ctx->devices));
+  ctx->currectDeviceIndex = deviceIndex;
+  ctx->deviceHandler = ctx->devices[deviceIndex];
+
+  // Get properties of the selected device
+  ZE_CHECK_ERROR(zeDeviceGetProperties(ctx->deviceHandler, &ctx->deviceProperties));
+  printf("Devicie initialized: Index: %d, Name: %s, Type: %d, VendorID: %x\n",
+         deviceIndex, ctx->deviceProperties.name, ctx->deviceProperties.type,
+         ctx->deviceProperties.vendorId);
+
+  // Create device command queue
+  ze_command_queue_desc_t l0_cq_desc = {
+      .stype = ZE_STRUCTURE_TYPE_COMMAND_QUEUE_DESC,
+      .pNext = NULL,
+      .ordinal = 0, /* this must be less than device_properties.numAsyncComputeEngines */
+      .index = 0,
+      .flags = 0,
+      .mode = ZE_COMMAND_QUEUE_MODE_DEFAULT,
+      .priority = ZE_COMMAND_QUEUE_PRIORITY_NORMAL};
+  ZE_CHECK_ERROR(zeCommandQueueCreate(ctx->deviceContext, ctx->deviceHandler, &l0_cq_desc,
+                                      &ctx->deviceCommandQueue));
+
+  // Create device command list
+  ze_command_list_desc_t l0_cl_desc = {.stype = ZE_STRUCTURE_TYPE_COMMAND_LIST_DESC,
+                                       .pNext = NULL,
+                                       .commandQueueGroupOrdinal = 0,
+                                       .flags = 0};
+  ZE_CHECK_ERROR(zeCommandListCreate(ctx->deviceContext, ctx->deviceHandler, &l0_cl_desc,
+                                     &ctx->deviceCommandList));
+
+  ctx->initialized = 1;
+  return 0;
+}
+
+/**
+ * @brief Allocate shared memory for the device.
+ *
+ * @param ctx - GPU context of the device.
+ * @param buf in/out - buf pointer to be filled
+ * @param size
+ * @return int. 0 if successfull, if error occured -1.
+ */
+int gpu_allocate_shared_buffer(GpuContext* ctx, void** buf, size_t size) {
+  // check if ctx is initialized
+  CTX_CHECK_INIT(ctx);
+
+  ze_device_mem_alloc_desc_t deviceMemDesc = {
+      .stype = ZE_STRUCTURE_TYPE_DEVICE_MEM_ALLOC_DESC,
+      .pNext = NULL,
+      .flags = 0,
+      .ordinal = 0 /* this must be less than count of zeDeviceGetMemoryProperties */
+  };
+  ze_host_mem_alloc_desc_t hostMemDesc = {
+      .stype = ZE_STRUCTURE_TYPE_HOST_MEM_ALLOC_DESC, .pNext = NULL, .flags = 0};
```

Comment:
```suggestion
      .stype = ZE_STRUCTURE_TYPE_HOST_MEM_ALLOC_DESC,
      .pNext = NULL,
      .flags = 0,
    };
```

Reply:
sorry coding standards with ./format_code.sh 
do not allow for this change the lint is flaging this :( 

Although i agree its more readable